server/src/main/java/io/druid/client/cache/MapCache.java

```diff
@@ -114,51 +121,25 @@ public void put(NamedKey key, byte[] value)
   @Override
   public void close(String namespace)
   {
-    byte[] idBytes;
-    synchronized (namespaceId) {
-      idBytes = getNamespaceId(namespace);
-      if (idBytes == null) {
-        return;
-      }
-
-      namespaceId.remove(namespace);
-    }
-    synchronized (clearLock) {
-      Iterator<ByteBuffer> iter = baseMap.keySet().iterator();
+    final Integer id = namespaceId.asMap().remove(namespace);
```

Comment:
That would have the potential to waste memory temporarily, since it would auto-recover after a time I agree its probably fine to just let the LRU get rid of it over time.